### PR TITLE
Fix #17397: Fully export instrument changes to MusicXML

### LINKED ISSUE #17397: [Musicxml Export] - Instrument change still not exported
### Issue type

Other type of issue

### Bug description

_No response_

### Steps to reproduce

When there's an instrument change with either or both long and short names, it must be displayed in Musicxml as part-name, part-abbreviation, or the other pair with display-text. Currently none is exported, only transposition (if have) is exported. This brings confusion in both importing and braille transcription.

### Screenshots/Screen recordings

_No response_

### MuseScore Version

4.0.2

### Regression

No.

### Operating system

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR adds full export of instrument changes to MusicXML support.

Closes #17397.

NB: Import currently doesn't change the instrument name yet, so this has to be adjested for fully working round-trips of import and export.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
As far as I can tell there's nothing to use/run those tests?

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -7731,6 +7732,48 @@ static void writeStaffDetails(XmlWriter& xml, const Part* part)
...
+    const Part* part = instrChange->part();
+    const size_t partNr = muse::indexOf(m_score->parts(), part);
+    const int instNr = muse::value(m_instrMap, instr, -1);
+    static const std::wregex acc(L"[♭♯]");
+
+    m_xml.startElement("print");
+    m_xml.tag("part-name", instr->nameAsPlainText().replace(u"♭", u"b").replace(u"♯", u"#"));
+    if (instr->nameAsPlainText().contains(acc)) {
+        m_xml.startElement("part-name-display");
+        writeDisplayName(m_xml, instr->nameAsPlainText());
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:7755]: you're calling `instr->nameAsPlainText()`3 times, I guess it'd be better to use a variable and call it just once

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 2]
```diff
@@ -7731,6 +7732,48 @@ static void writeStaffDetails(XmlWriter& xml, const Part* part)
...
+        m_xml.endElement();
+    }
+    if (!instr->abbreviatureAsPlainText().isEmpty()) {
+        m_xml.tag("part-abbreviation", instr->abbreviatureAsPlainText().replace(u"♭", u"b").replace(u"♯", u"#"));
+        if (instr->abbreviatureAsPlainText().contains(acc)) {
+            m_xml.startElement("part-abbreviation-display");
+            writeDisplayName(m_xml, instr->abbreviatureAsPlainText());
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:7762]: same here

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 3]
```diff
@@ -7731,6 +7732,48 @@ static void writeStaffDetails(XmlWriter& xml, const Part* part)
...
+            m_xml.endElement();
+        }
+    }
+    m_xml.endElement();
+
+    writeInstrumentDetails(instrChange->instrument(), m_score->style().styleB(Sid::concertPitch));
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:7768]: why not using `instr` here too?

Comment (rettinghaus) [reply src/importexport/musicxml/internal/musicxml/exportxml.cpp]: Thanks for your review! Actually found another thing that lead to invalid MusicXML output. Fixed that one, too.

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 4]
```diff
@@ -7731,6 +7732,46 @@ static void writeStaffDetails(XmlWriter& xml, const Part* part)
...
+
+void ExportMusicXml::writeInstrumentChange(const InstrumentChange* instrChange)
+{
+    const Instrument* instr = instrChange->instrument();
+    const Part* part = instrChange->part();
+    const size_t partNr = muse::indexOf(m_score->parts(), part);
+    const int instNr = muse::value(m_instrMap, instr, -1);
+    const string longName = instr->nameAsPlainText();
+    const string shortName = instr->abbreviatureAsPlainText();
+    static const std::wregex acc(L"[♭♯]");
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:7751]: Unused variable?

**Review by miiizen — APPROVED** [review]

**Pull request merged**